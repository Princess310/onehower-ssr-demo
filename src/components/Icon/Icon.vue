<template>
  <div>
    <svg :width="size" :height="size" :color="color">
      <use :xlink:href="`#${type.default.id}`" />
    </svg>
  </div>
</template>

<script>
export default {
  name: 'oh-icon',
  props: {
    type: Object,
    size: {
      type: String,
      default: '24px',
    },
    color: {
      type: String,
      default: '#cccccc',
    },
  },
};
</script>

<style scoped>
</style>
