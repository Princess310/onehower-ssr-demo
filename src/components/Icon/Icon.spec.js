import { shallow } from 'vue-test-utils'
import Icon from '@/components/Icon';

describe('Icon.vue', () => {
  it('should set correct size and color', () => {
    const wrapper = shallow(Icon, {
      propsData: {
        size: '32px',
        color: 'green',
        // type prop is control by svg-sprite-loader like require('../../icons/github.svg'),
        type: {
          default: {
            id: 'test',
          },
        },
      },
    });

    const attributes = wrapper.find('svg').attributes();
    expect(attributes.color).toBe('green');
    expect(attributes.width).toBe('32px');
  });

  it('should match a snapshot', () => {
    const wrapper = shallow(Icon, {
      propsData: {
        // type prop is control by svg-sprite-loader like require('../../icons/github.svg'),
        type: {
          default: {
            id: 'test',
          },
        },
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  }) 
});
