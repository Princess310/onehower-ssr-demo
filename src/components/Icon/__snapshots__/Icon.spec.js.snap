// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon.vue should match a snapshot 1`] = `
<div>
    <svg width="24px" height="24px" color="#cccccc">
        <use xlink:href="#test"></use>
    </svg>
</div>
`;
